### PR TITLE
syphilis test card support backend

### PR DESCRIPTION
**Pull request by fzhao99**
# BACKEND PULL REQUEST

## Related Issue

- Backend work for #7436 that modifies the AoE storage to accept the new syphilis history question

## Changes Proposed

- Modifies the AoE backend object to CRUD a new syphilis history field needed for the new card
- Updates the relevant GQL endpoints to query / mutate this field
- Refactors the tests with a builder pattern / with the new field for readability

## Additional Information
- I decided just to add the new field onto the existing AoE JSON like we did for the other questions, but worth considering how we refactor these endpoints in the future to be more flexible to adding new fields. Had to change the value in several places in order to get everything working, which may be an issue as we add additional sets of AoE questions for the new STI's that are coming down the pike
- I didn't add any validation for the validity of the syphilis type codes just yet because 1) we don't do that for pregnancy and 2) we're waiting for confirmation from LAC that the SNOMED's I found on Google were correct, but we might want to add a check that the values match.

Comments on either of the two above decisions welcome

## Testing

https://github.com/CDCgov/prime-simplereport/assets/29645040/c18222f0-c770-4fdf-a924-60483436ef2c

- Grab a facility ID / patient ID combination in your local dev. 
- Use the below queries to add a patient to the queue 

```gql
// Variables for mutation
{
  "patientId": YOUR-PATIENT-ID,
  "noSymptoms": true,
  "symptoms": "{\"25064002\":false,\"36955009\":false,\"43724002\":false,\"44169009\":false,\"49727002\":false,\"62315008\":false,\"64531003\":false,\"68235000\":false,\"68962001\":false,\"84229001\":false,\"103001002\":false,\"162397003\":false,\"230145002\":false,\"267036007\":false,\"422400008\":false,\"422587007\":false,\"426000000\":false}",
  "pregnancy": "60001007",
  "syphilisHistory": "14732006", // or a test string
  "testResultDelivery": null,
  "facilityId": YOUR-FACILITY-ID
}

mutation AddPatientToQueue($facilityId: ID!, $patientId: ID!, $symptoms: String, $symptomOnset: LocalDate, $pregnancy: String, $noSymptoms: Boolean, $syphilisHistory: String, $testResultDelivery: TestResultDeliveryPreference) {
  addPatientToQueue(
    facilityId: $facilityId
    patientId: $patientId
    pregnancy: $pregnancy
    syphilisHistory: $syphilisHistory
    noSymptoms: $noSymptoms
    symptoms: $symptoms
    symptomOnset: $symptomOnset
    testResultDelivery: $testResultDelivery
  )
}
```
- Use the below query to grab the queue back and notice the presence of the syphilis history code
```gql
query Queue ($facilityId: ID!) {
      queue (
        facilityId: $facilityId
      ){
        internalId
        pregnancy
        dateAdded
        syphilisHistory
      }
    }
```

---

### CODE REVIEW

File: backend/src/main/java/gov/cdc/usds/simplereport/db/model/PersonUtils.java [hunk 1]
```diff
@@ -795,6 +795,11 @@ public static final Map<String, String> getResidenceTypeMap() {
   private static final String PREGNANT_SNOMED = "77386006";
   private static final String NOT_PREGNANT_SNOMED = "60001007";
   private static final String PREGNANT_UNKNOWN_SNOMED = "102874004";
+
+  public static final String YES_SYPHILIS_HISTORY_SNOMED = "1087151000119108";
```
Comment (fzhao99) [inline backend/src/main/java/gov/cdc/usds/simplereport/db/model/PersonUtils.java:799]: put a q out via Jayna for LAC to make sure these are the codes we should use. Will update these as necessary as that answer gets back to us.

File: backend/src/test/java/gov/cdc/usds/simplereport/api/converter/FhirConverterTest.java
Comment (bobbywells52) [inline backend/src/test/java/gov/cdc/usds/simplereport/api/converter/FhirConverterTest.java:1]: The refactor makes this so much easier to see what the tests are doing -- thanks for this!

File: backend/src/main/java/gov/cdc/usds/simplereport/api/queue/QueueMutationResolver.java [hunk 2]
```diff
@@ -90,6 +92,7 @@ public void removePatientFromQueue(@Argument UUID patientId) {
   public void updateTimeOfTestQuestions(
```
[the remark this answers is not in the document]
Comment (bobbywells52) [reply backend/src/main/java/gov/cdc/usds/simplereport/api/queue/QueueMutationResolver.java:92]: I second the need for a rename -- was also confused by the naming convention

**Review by bobbywells52 — APPROVED** [review]
Tested locally and it's working for me -- thanks for your work on this! From what I can tell you didn't miss updating any of the graphQL endpoints that needed to be updated. However, given how many that there are would it make sense for the team to compile a checklist of all the endpoints that need to be updated on this type of a PR for adding a new disease test card?

**Review by emyl3 — APPROVED** [review]
LGTM! tested locally

@fzhao99 From my understanding, the backend and frontend changes can go in at the same time because there shouldn't be any issues with backwards compatibility for users with an out of date frontend based on these changes.

**Pull request merged**